### PR TITLE
fix(config types): inline types in config.d.ts

### PR DESCRIPTION
**Pull request by schultzp2020**
fixes https://github.com/janus-idp/backstage-plugins/issues/656
fixes https://github.com/janus-idp/backstage-plugins/issues/657
fixes https://github.com/janus-idp/backstage-plugins/issues/658
fixes https://github.com/janus-idp/backstage-plugins/issues/645
fixes https://github.com/janus-idp/backstage-plugins/issues/655
fixes minor config issues in `analytics-provider-segment`

### CODE REVIEW

File: plugins/analytics-provider-segment/config.d.ts [hunk 1]
```diff
@@ -20,26 +20,47 @@ export interface Config {
...
-         * to false.
-         * @visibility frontend
-         */
-        testMode?: boolean;
+            /**
+             * Prevents events from actually being sent when set to true. Defaults
+             * to false.
+             * @visibility frontend
+             */
+            testMode: true;
```
Comment (tumido) [inline plugins/analytics-provider-segment/config.d.ts:36]: Why can't we leave it `testMode?`

File: yarn.lock [hunk 2]
```diff
@@ -4772,17 +4772,6 @@
...
-  version "1.3.0"
-  resolved "https://registry.yarnpkg.com/@janus-idp/shared-react/-/shared-react-1.3.0.tgz#c945834ee77548ad0336f9f7b61b1e6d85cdcbe6"
-  integrity sha512-ulnSgcE/zyq/F0RYY+mZTY6kC7rug8BDJG82+xFowkhj6TtduvzIa63FBDD685pUXI5lPfqiAvxCVOnBTDx7tQ==
-  dependencies:
-    "@kubernetes/client-node" "^0.18.1"
-    classnames "2.3.2"
-    date-fns "^2.30.0"
-    lodash "^4.17.19"
-    mathjs "^11.9.1"
-
```
Comment (tumido) [inline yarn.lock:4785]: Interesting.. Do we have `yarn.lock` out of sync?

File: plugins/analytics-provider-segment/config.d.ts (hunk 1 above)
Comment (schultzp2020) [reply plugins/analytics-provider-segment/config.d.ts:36]: If we left it as `testMode?: true` that would intersect with `testMode?: false`. For a complete union that does not intersect we want `testMode: true` and `testMode?: false`. i.e. `true` and `false | undefined` -> `boolean | undefined`.

File: yarn.lock (hunk 2 above)
Comment (schultzp2020) [reply yarn.lock:4785]: Do you want me to remove `yarn.lock`?

Comment (tumido) [reply yarn.lock:4785]: Let's keep it if we're out of sync... 🤷

**Review by tumido — APPROVED** [review]
/lgtm
/approve

**Comment by invincibleJai** [comment]
LGTM

File: yarn.lock (hunk 2 above)
Comment (schultzp2020) [reply yarn.lock:4785]: @tumido

Comment (schultzp2020) [reply yarn.lock:4785]: @nickboldt we will need to build `shared-react` downstream since it is used in many of our plugins.

**Review by fmenesesg — APPROVED** [review]
/lgtm

File: yarn.lock (hunk 2 above)
Comment (nickboldt) [reply yarn.lock:4785]: ok, good, I've been doing that for a month now :D

**Pull request merged**